File: 1537-maximum-score-after-splitting-a-string/maximum-score-after-splitting-a-string.py
class Solution:
    def maxScore(self, s: str) -> int:
        n = len(s)
        mx = 0

        for i in range(1, n):
            left = s[0:i]
            right = s[i:n]

            if left.count("0") > 0 or right.count("1") > 0:
                zero = left.count("0")
                one = right.count("1")

                mx = max(mx, zero + one)

        return mx

File: 1537-maximum-score-after-splitting-a-string/maximum-score-after-splitting-a-string.js
/**
 * @param {string} s
 * @return {number}
 */
var maxScore = function (s) {
    const n = s.length;
    let maxScore = 0;

    for (let i = 1; i < n; i++) {
        let left = s.substring(0, i);
        let right = s.substring(i, n);

        if (left.split("0").length - 1 > 0 || right.split("1").length - 1 > 0) {
            let zero = left.split("0").length - 1;
            let one = right.split("1").length - 1;

            maxScore = Math.max(maxScore, zero + one);
        }
    }

    return maxScore;
};